src/png.imageio/pngoutput.cpp

```diff
@@ -289,11 +310,12 @@ bool
 PNGOutput::write_scanline(int y, int z, TypeDesc format, const void* data,
                           stride_t xstride)
 {
+    bool need_swap = (m_spec.format == TypeDesc::UINT16 && littleendian());
```

Comment:
Why not use `m_need_swap` here in this method, and below in `PNGOutput::write_scanlines`?